### ISSUE TITLE
Enhance Wildfly & EAP port detection

### ISSUE THREAD

**Issue by thepetk** [redhat-developer/alizer#195]
## Which area this feature is related to?

/area cli

## Issue details

After the implementation of [Wildfly and JBoss EAP detectors](https://github.com/redhat-developer/alizer/pull/161) we need to improve the port detection for these cases.

The first attempt to do that, was to detect a `standalone.xml` file and check its contents (see more info [here](https://github.com/redhat-developer/alizer/pull/161#discussion_r1158210346)). But as this file is produced after a maven build we are not able to detect ports from it.

Another workaround could be to check the `pom.xml` file for any scripts that are might configuring the server. An example inside pom.xml could be:
```xml
<packaging-scripts>
    <packaging-script>
      <scripts>
        <script>../scripts/script1.cli</script>
      </scripts>
    </packaging-script>
<packaging-scripts>
```
An example line that could configure a port could be (Check more information [here](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/configuration_guide/network_and_port_configuration#socket_bindings)):
```
/socket-binding-group=new-sockets/socket-binding=new-socket-binding:add(port=1234)
```

Example project using cli file: https://github.com/wildfly/quickstart/tree/main/servlet-security

**Target Date**: 26 June 2023

**Comment by thepetk** [comment]
Started working on the proposed workaround :)

**Comment by thepetk** [comment]
I've added this [PR](https://github.com/redhat-developer/alizer/pull/219) for wildfly port detection based on `packaging-script`. The next step is to introduce a JBoss EAP port detection following [this resource](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.1/html/management_cli_guide/config_management_cli)

**Comment by thepetk** [comment]
I've updated all test cases after the merge of #206 and both PRs #221 and #219  are ready for review.

@ehsavoie I have added you in both PRs as reviewer regarding the logic of port detection we are introducing for Wildfly and JBoss EAP :)

**Comment by thepetk** [comment]
After further investigation, I've decided to update the detection logic and detect ports from the configuration of the `wildfly-maven-plugin`. An example of this configuration would be:

```xml
<plugin>
    <groupId>org.wildfly.plugins</groupId>
    <artifactId>wildfly-maven-plugin</artifactId>
    <version>4.1.0.Final</version>
    <configuration>
        <javaOpts>-Djboss.http.port=8380</javaOpts>
    </configuration>
</plugin>
```
More info here: https://docs.wildfly.org/wildfly-maven-plugin/

**Comment by thepetk** [comment]
After a fair investigation for the best solution, we have concluded to use:

* For wildfly:
```xml
<plugin>
    <groupId>org.wildfly.plugins</groupId>
    <artifactId>wildfly-maven-plugin</artifactId>
    <version>x.x.x</version>
    <configuration>
        <javaOpts>-Djboss.http.port=8380</javaOpts>
    </configuration>
</plugin>
```

* For jboss-eap:
```xml
<plugin>
    <groupId>org.jboss.eap.plugins</groupId>
    <artifactId>eap-maven-plugin</artifactId>
    <version>x.x.x</version>
    <configuration>
        <javaOpts>-Djboss.http.port=8380</javaOpts>
    </configuration>
</plugin>
```

Additionally an issue was created for a feature request regarding the [wildfly-jar-maven-plugin](https://github.com/wildfly-extras/wildfly-jar-maven-plugin): #243

**Comment by thepetk** [comment]
The [Jboss port detection PR](#221) has been merged, the [wildfly one](#219) is rebased under currently under review. After its merge the issue will be closed.
